resources/process-autogenerated-form/src/js/app.js

```diff
@@ -83,14 +83,36 @@
         $scope.postData = function postData() {
             $scope.message = undefined;
             var jsonifiedDataToSend = jsonify($scope.dataToSend);
-            contractSrvc.startProcess(processId, jsonifiedDataToSend).then(function () {
-
-                $window.top.location.href = '/bonita';
-            }, function (reason) {
-                $scope.message = reason.data.explanations ? reason.data.explanations : reason.data.message;
-            });
+            contractSrvc.startProcess(processId, jsonifiedDataToSend).then(
+                onPostSuccess, onPostError);
         };
 
+        var onPostSuccess = function(response) {
+            //if the form is displayed in an iframe
+            if ($window.parent !== $window.self) {
+                notifyParentFrame('success', response.status);
+            } else {
```

Comment:
I thought so too initially, but what happens if a user designed his own task list and display the forms in an iframe ? I'm pretty sure he doesn't want to get the portal displayed in his iframe. I think it's better to leave him the choice (he can listen to the messages posted and decide to redirect or not). This means we need to document the postMessage.
I think it's either that or we display a confirmation message instead of trying to redirect...